### PR TITLE
Assign current iteration to issue card

### PR DESCRIPTION
**Pull request by rodrigo-lourenco-lopes**
## Description

This workflow assigns the current iteration to the issue card once the issue gets moved to the current iteration board.

### CODE REVIEW

**Comment by rodrigo-lourenco-lopes** [comment]
Ran the workflow locally with [act](https://github.com/nektos/act#configuration) 
Using the command `act -j assign_lable -W .github/workflows/assign-current-iteration-to-issue.yml --container-architecture linux/amd64` (the last part only necessary for arm64 users)
Although this test only guaranteed the semantic correctness of the workflow.

File: .github/workflows/assign-current-iteration-to-issue.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    runs-on: ubuntu-latest
+    permissions:
+      issues: write
+      pull-requests: write
+    steps:
+      - uses: blombard/move-to-current-iteration@master
+        with:
+          owner: camunda
+          number: 92 # project number
+          token: ${{secrets.GITHUB_TOKEN}}
```
Comment (megglos) [inline .github/workflows/assign-current-iteration-to-issue.yml:21]:
```suggestion
          token: ${{ secrets.PROJECT_ADMIN_TOKEN }}
```

File: .github/workflows/assign-current-iteration-to-issue.yml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+          iteration-field: Iteration
+          iteration: last
```
Comment (megglos) [inline .github/workflows/assign-current-iteration-to-issue.yml:23]: what is the meaning of this config, it seems like a precondition? I guess then it won't be met if we move from inbox to current?

**Review by megglos — DISMISSED** [review]
I'm not sure it will do what we want yet:
- Set the iteration to `current` when an issue is moved into the `Current Iteration` status, let's sync

**Comment by npepinpe** [comment]
@rodrigo-lourenco-lopes if you get me up to speed on what's missing we can merge this :+1:

File: .github/workflows/assign-current-iteration-to-issue.yml [hunk 3]
```diff
@@ -0,0 +1,24 @@
+# This workflow assigns the current iteration to issue card once the issue gets moved to
+# the current iteration board.
+
+name: assign current iteration
+on:
+  project_card:
+    types:
+      - moved
+jobs:
+  assign_lable:
```
Comment (deepthidevaki) [inline .github/workflows/assign-current-iteration-to-issue.yml:10]:
```suggestion
  assign_label:
```

File: .github/workflows/assign-current-iteration-to-issue.yml [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+    if: github.event.project_card.column_id == '47fc9ee4' # this id can be found via the inspection tool on the browser
```
Comment (deepthidevaki) [inline .github/workflows/assign-current-iteration-to-issue.yml:11]: :question: Should we also filter for the expected project?

**Review by deepthidevaki — APPROVED** [review]
Thanks Rodrigo. Assuming the used action works as expected, I'm approving this PR so that you can test is after merging.

File: .github/workflows/assign-current-iteration-to-issue.yml (hunk 4 above)
Comment (rodrigo-lourenco-lopes) [reply .github/workflows/assign-current-iteration-to-issue.yml:11]: In the documentation, they also just check for the column ID in a similar example. I assume it should not be possible to have an ID conflict within a repo.
https://docs.github.com/en/actions/managing-issues-and-pull-requests/removing-a-label-when-a-card-is-added-to-a-project-board-column

**Comment by rodrigo-lourenco-lopes** [comment]
bors merge

**Comment by rodrigo-lourenco-lopes** [comment]
bors retry

**Pull request merged**